dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,721 @@
+import org.apache.commons.lang3.mutable.MutableObject;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.io.StringWriter;
+import java.security.NoSuchAlgorithmException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Future;
+
+/**
+ * Represents a Storage on the database It supports big files since provides the ability to split
+ * fat object in smaller pieces, see {@link FileByteSplitter} and {@link com.dotcms.util.FileJoiner}
+ * to get more details about the process
+ *
+ * @author jsanca
+ */
+public class DataBaseStoragePersistenceAPIImpl implements StoragePersistenceAPI {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * custom external connection provider method
+     * in case we want to store stuff outside our db
+     * @return
+     */
+    protected Connection getConnection() {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        return isExternalPool ?
+                DbConnectionFactory.getConnection(jdbcPool) :
+                DbConnectionFactory.getConnection();
+    }
+
+    /**
+     * custom close connection wrapper
+     * @param voidDelegate
+     */
+    protected void wrapCloseConnection(final VoidDelegate voidDelegate) {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+
+            wrapExternalCloseConnection(voidDelegate, jdbcPool);
+        } else {
+
+            wrapLocalCloseConnection(voidDelegate);
+        }
+    }
+
+    /**
+     * custom close connection wrapper
+     * @param voidDelegate
+     */
+    private void wrapLocalCloseConnection(final VoidDelegate voidDelegate) {
+
+        final boolean isNewConnection = !DbConnectionFactory.connectionExists();
+
+        try {
+
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            if (isNewConnection) {
+
+                DbConnectionFactory.closeSilently();
+            }
+        }
+    }
+
+    /**
+     * custom external close connection wrapper
+     * @param voidDelegate
+     * @param jdbcPool
+     */
+    private void wrapExternalCloseConnection(final VoidDelegate voidDelegate,
+            final String jdbcPool) {
+
+        Connection connection = null;
```

Comment:
Codacy found an issue: [Ensure that resources like this Connection object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)